File: problems/1. twoSum/twoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        itemsIndex = dict()
        for index, i in enumerate(nums):
            if i in itemsIndex:
                itemIndex = itemsIndex[i]
                return [itemIndex, index]
            
            result = target - i
            itemsIndex[result] = index

        return None

File: problems/1. twoSum/twoSum.ts
function twoSum(nums: number[], target: number): number[] | null {
  const itemsIndex = new Map<number, number>();

  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    if (itemsIndex.has(num)) {
      return [itemsIndex.get(num)!, i];
    }
    const result = target - num;
    itemsIndex.set(result, i);
  }
  return null;
}
